web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -0,0 +1,204 @@
+import io.deephaven.javascript.proto.dhinternal.arrow.flight.flatbuf.schema_generated.org.apache.arrow.flatbuf.Field;
+import io.deephaven.javascript.proto.dhinternal.arrow.flight.flatbuf.schema_generated.org.apache.arrow.flatbuf.Schema;
+import io.deephaven.javascript.proto.dhinternal.flatbuffers.ByteBuffer;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.partitionedtable_pb.GetTableRequest;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.partitionedtable_pb.MergeRequest;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.partitionedtable_pb.PartitionedTableDescriptor;
+import io.deephaven.web.client.api.barrage.BarrageUtils;
+import io.deephaven.web.client.api.barrage.def.ColumnDefinition;
+import io.deephaven.web.client.api.subscription.SubscriptionTableData;
+import io.deephaven.web.client.api.subscription.TableSubscription;
+import io.deephaven.web.client.api.widget.JsWidget;
+import io.deephaven.web.client.state.ClientTableState;
+import io.deephaven.web.shared.data.RangeSet;
+import jsinterop.annotations.JsIgnore;
+import jsinterop.annotations.JsProperty;
+import jsinterop.annotations.JsType;
+import jsinterop.base.Any;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@JsType(namespace = "dh", name = "PartitionedTable")
+public class JsPartitionedTable extends HasEventHandling {
+    public static final String EVENT_KEYADDED = "keyadded",
+            EVENT_DISCONNECT = JsTable.EVENT_DISCONNECT,
+            EVENT_RECONNECT = JsTable.EVENT_RECONNECT,
+            EVENT_RECONNECTFAILED = JsTable.EVENT_RECONNECTFAILED;
+
+    private final WorkerConnection connection;
+    private final JsWidget widget;
+    private List<String> keyColumnTypes;
+    private PartitionedTableDescriptor descriptor;
+    private JsTable keys;
+    private TableSubscription subscription;
+
+    /*
+     * Represents the sorta-kinda memoized results, tables that we've already locally fetched from the tablemap, and if
+     * all references to a table are released, entries here will be replaced with unresolved instances so we don't leak
+     * server references or memory. Keys are Object[], even with one element, so that we can easily hash without an
+     * extra wrapper object. Since columns are consistent with a PartitionedTable, we will not worry about "foo" vs
+     * ["foo"] as being different entries.
+     */
+    private final Map<JsArray<Any>, JsLazy<Promise<ClientTableState>>> tables = new HashMap<>();
+
+
+    @JsIgnore
+    public JsPartitionedTable(WorkerConnection connection, JsWidget widget) {
+        this.connection = connection;
+        this.widget = widget;
+    }
+
+    @JsIgnore
+    public Promise<JsPartitionedTable> refetch() {
+        if (keys != null) {
+            keys.close();
```

Comment:
refetch() isn't exported, and should only be called internally - we don't offer canceling refetch in other cases. The purpose of refetch is to be able to gracefully handle being reconnected, rather than getting the data again or something.